src/vaults/yields.utils.ts

```diff
@@ -421,12 +423,22 @@ async function evaluateYieldEvents(chain: Chain, vault: VaultDefinitionModel): P
   // this error should bubble up to yield source persistence
   // from a practical perspective this is here to allow vaults to retain calculated yield if there is no history
   // this is primary needed for a clean resync of harvest data without impacting 'synced' data
-  if (yieldEvents.length === 0) {
-    throw new Error(`${vault.name} has no recent harvests, it is either not synced, or effectively deprecated`);
+  if (vault.state !== VaultState.Discontinued && yieldEvents.length === 0) {
```

Comment:
do not update non discontinued vault aprs while syncing - we can consider adding error reporting here soon